client.go

```diff
@@ -151,19 +151,7 @@ func (c *Client) checkResponse(ctx context.Context, retryCount int, resp *http.R
 			}
 		}
 	}
-	if resp.StatusCode == http.StatusPreconditionFailed {
-		return ErrPreconditionFailed
-	}
-	if resp.StatusCode == http.StatusConflict {
-		return ErrConflict
-	}
-	if resp.StatusCode >= 300 {
-		err := &RequestError{}
-		readJson(resp.Body, &err)
-		return err
-	}
-
-	return nil
+	return CosmosHTTPErrors[resp.StatusCode]
```

Comment:
should we check that the map returned something? If we get a status code that's not covered, we will return `nil` atm.

something like
```
comsosError, ok := CosmosHttpErrors[resp.StatusCode]
if !ok {
    // return something static
}
```

Reply:
True, thats not good. I will return a "Unexpected return status" or similar for now